### PR TITLE
Leave Guild

### PR DESCRIPTION
**Pull request by AstralSkies**
- Implemented the leave guild handler.
- Implemented the associated dialog box w/o the sound effect.
- Implemented the dialog boxes associated with grabbing information etc, guild look up, view member.

### CODE REVIEW

File: EndlessClient/Dialogs/GuildDialog.cs [hunk 1]
```diff
@@ -13,64 +14,112 @@
...
         private GuildDialogState _state;
+        private DialogAction _currentDialogAction;
 
         public GuildDialog(INativeGraphicsManager nativeGraphicsManager,
                          IEODialogButtonService dialogButtonService,
                          IEODialogIconService dialogIconService,
                          ILocalizedStringFinder localizedStringFinder,
                          ITextInputDialogFactory textInputDialogFactory,
-                         IGuildActions GuildActions,
+                         IGuildActions guildActions,
```
Comment (AstralSkies) [inline EndlessClient/Dialogs/GuildDialog.cs:70]: opps

File: EOLib/PacketHandlers/Guild/GuildLeaveHandler.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+using EOLib.Net.Handlers;
+using Moffat.EndlessOnline.SDK.Protocol.Net;
+using Moffat.EndlessOnline.SDK.Protocol.Net.Server;
+using Optional;
+using System.Collections.Generic;
+
+namespace EOLib.PacketHandlers.Guild
+{
+    [AutoMappedType]
+    public class GuildLeaveHandler : InGameOnlyPacketHandler<GuildKickServerPacket>
```
Comment (ethanmoffat) [inline EOLib/PacketHandlers/Guild/GuildLeaveHandler.cs:15]: There a few handlers that don't follow this, but can you rename this to `GuildKickHandler` to match the packet id? I've been trying to update handler names to this convention when I see they're out of sync.

File: EOLib/PacketHandlers/Guild/GuildLeaveHandler.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            : base(playerInfoProvider)
+        {
+            _npcInteractionNotifiers = npcInteractionNotifiers;
+        }
+
+        public override bool HandlePacket(GuildKickServerPacket packet)
+        {
+
+            foreach (var notifier in _npcInteractionNotifiers)
+                notifier.NotifyInteractionFromNPC(IO.NPCType.Guild);
```
Comment (ethanmoffat) [inline EOLib/PacketHandlers/Guild/GuildLeaveHandler.cs:34]: Sending this doesn't seem right for leaving a guild. The intent of these notifiers is to signal a packet was received when trying to interact with a map NPC - the dialog doesn't open until the packet is received and the client is 'notified'. I don't think that intent should be broken/changed here.

File: EndlessClient/Dialogs/GuildDialog.cs [hunk 4]
```diff
@@ -13,64 +14,112 @@
...
+                         ISfxPlayer sfxPlayer)
             : base(nativeGraphicsManager, dialogButtonService, DialogType.Guild)
         {
+            _nativeGraphicsManager = nativeGraphicsManager;
+            _dialogButtonService = dialogButtonService;
             _dialogIconService = dialogIconService;
             _localizedStringFinder = localizedStringFinder;
             _textInputDialogFactory = textInputDialogFactory;
-            _GuildActions = GuildActions;
+            _GuildActions = guildActions;
```
Comment (ethanmoffat) [inline EndlessClient/Dialogs/GuildDialog.cs:85]: You can lower case the field name too

File: EndlessClient/Dialogs/GuildDialog.cs [hunk 5]
```diff
@@ -13,64 +14,112 @@
...
                          ILocalizedStringFinder localizedStringFinder,
                          ITextInputDialogFactory textInputDialogFactory,
-                         IGuildActions GuildActions,
+                         IGuildActions guildActions,
                          IContentProvider contentProvider,
                          ICurrentMapStateProvider currentMapStateProvider,
-                         IENFFileProvider enfFileProvider)
+                         IENFFileProvider enfFileProvider,
+                         IEOMessageBoxFactory messageBoxFactory,
+                         ICharacterRepository characterLvRepository,
```
Comment (ethanmoffat) [inline EndlessClient/Dialogs/GuildDialog.cs:75]: `characterLvRepository` -> `characterRepository`

File: EndlessClient/Dialogs/GuildDialog.cs [hunk 6]
```diff
@@ -128,15 +183,173 @@ private void SetState(GuildDialogState state)
...
+                    break;
+            }
+        }
+
+        private void HandleDialogAction(string responseText)
+        {
+            switch (_currentDialogAction)
+            {
+                case DialogAction.GuildLookUp:
+                    Debug.WriteLine("Call Guild look up");
```
Comment (ethanmoffat) [inline EndlessClient/Dialogs/GuildDialog.cs:301]: If you're not familiar with the debugger, I'll write up a quick tutorial on helpful tips for using it. Scattering `Debug.WriteLine` everywhere through the code is not a sustainable solution

**Review by ethanmoffat — COMMENTED** [review]
Please update the PR title to a short summary of what this change did (I think it's mostly the "leave a guild" function?)

File: EndlessClient/Dialogs/GuildDialog.cs (hunk 6 above)
Comment (AstralSkies) [reply EndlessClient/Dialogs/GuildDialog.cs:301]: Thanks! `Debug.WriteLine` is usually my go-to, and I'm somewhat familiar with the debugger, but a quick tutorial would still be very helpful. Appreciate it!

**Pull request closed without merging**